libs/core-ui/src/lib/util/calculateAUCData.ts

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { localization } from "@responsible-ai/localization";
+import { SeriesOptionsType } from "highcharts";
+import { range, unzip } from "lodash";
+
+import { IDataset } from "../Interfaces/IDataset";
+
+interface IPoint {
+  x: number;
+  y: number;
+}
+export interface IROCData {
+  points: IPoint[];
+}
+
+function getStaticROCData(): SeriesOptionsType[] {
+  return [
+    {
+      data: [
+        { x: 0, y: 0 },
+        { x: 0, y: 1 },
+        { x: 1, y: 1 }
+      ],
+      name: localization.Interpret.Charts.Ideal,
+      type: "line"
+    },
+    {
+      data: [
+        { x: 0, y: 0 },
+        { x: 1, y: 1 }
+      ],
+      name: localization.Interpret.Charts.Random,
+      type: "line"
+    }
+  ];
+}
+
+export function calculatePerClassROCData(
+  probabilityY: number[],
+  binY: number[]
+): IROCData {
+  const rocData: IROCData = {
+    points: []
+  };
+  const thresholds = range(0, 1, 0.01);
```

Comment:
That's not quite right, is it?
For example, if I have two points within a tick (let's say at 0.001 and 0.004) then none of my thresholds covers the case where they are on opposite sides of the decision boundary. This could very well be the best FPR/TPR combination....

So I would order the values first, and then always take thresholds as the middle between two values, e.g.,

values: [1,5,4,2,3]
ordered: [1,2,3,4,5]
thresholds: [-inf, 1.5, 2.5, 3.5, 4.5, inf]
so total of 6 points for ROC, but then you still need to calculate the convex hull of those points if I'm not mistaken. How are you avoiding hitting points below the convex hull btw?